tools/scanner/scan.mk

```diff
@@ -0,0 +1,66 @@
+# Copyright 2020-2022 The MLPerf Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+# Note:
+# SonarScanner requires a clean build for every run. Do not cache build files.
```

Comment:
Why do we need a clean build?
Why do we even need a build for static analysis tool?

The issue I have with this is time.
Currently our CI is usually 30-35 minutes long, and I think it's already really slow (I wish we could make Windows CI faster).
From what I see in this PR checks, Sonar scan takes 50 minutes, roughly 1.5 times longer.
I think it's unacceptably long for a pull request check.

Even worse is that local checks will also take a long time.

Reply:
If it's impossible to reduce time for `build + sonar checks`, we can just disable it for pull requests but keep for the master branch commits.

Reply:
> Why do we need a clean build?
Why do we even need a build for static analysis tool?

It stated in the installation guide.

> The issue I have with this is time.
>  just disable it for pull requests but keep for the master branch commits.

Maybe we can make it optional and not required to merge, but I prefer to let it run on PR so I can check and fix the code if I'd like to.